android/capacitor/src/main/java/com/getcapacitor/WebViewLocalServer.java

```diff
@@ -294,22 +295,32 @@ private WebResourceResponse handleProxyRequest(WebResourceRequest request, PathH
         String path = request.getUrl().getPath();
         URL url = new URL(request.getUrl().toString());
         Map<String, String> headers = request.getRequestHeaders();
-        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-        for (Map.Entry<String, String> header : headers.entrySet()) {
-          conn.setRequestProperty(header.getKey(), header.getValue());
-        }
-        conn.setRequestMethod(method);
-        conn.setReadTimeout(30 * 1000);
-        conn.setConnectTimeout(30 * 1000);
+
+        HttpURLConnection conn;
+        boolean isContentTypeNull;
+        do {
```

Comment:
I'm not really confortable putting a while with Thread.sleep, might be useful for reconnection, but could cause problems if the networks is down.

Reply:
If there is a better way you could point me, I'll change it.

Reply:
Maybe I could throw an error after 10 attempts